### PR TITLE
Add a Nappy provider DAG using ProviderDataIngester

### LINKED ISSUE #704: Nappy |  Beautiful photos of Black and Brown people, for free.
### Provider API Endpoint / Documentation

https://api.nappy.co/v1/openverse/images

### Provider description

From https://nappy.co/why: 

> And that’s why we launched nappy; to provide beautiful, high-res photos of black and brown people to startups, brands, agencies, and everyone else. Nappy makes it easy for companies to be purposeful about representation in their designs, presentations, and advertisements.

Nappy is already in use as an 'additional source' on the frontend, so this deeper integration would be excellent. 

### Licenses Provided

[CC0](https://creativecommons.org/publicdomain/zero/1.0/)

### Provider API Technical info

Nappy does not have a documented API but is working with us to create an endpoint for use. 

1. GET [https://api.nappy.co/v1/openverse/images](https://href.li/?https://api.nappy.co/v1/openverse/images) (listing all images, paginated)
a. queries available (example):
a1. page=5 (current page, default: 1)
a2. per_page=3 (3 results per page, default: 10)
2. GET [https://api.nappy.co/v1/openverse/images/:id](https://href.li/?https://api.nappy.co/v1/openverse/images/:id) (specific image, where :id is the foreign_identifier and a number)

### Checklist to complete before beginning development

- [ ] Verify there is a way to retrieve the entire relevant portion of the provider's collection in a systematic way via their API.
- [ ] Verify the API provides license info (license type and version; license URL provides both, and is preferred)
- [ ] Verify the API provides stable direct links to individual works.
- [ ] Verify the API provides a stable landing page URL to individual works.
- [ ] Note other info the API provides, such as thumbnails, dimensions, attribution info (required if non-CC0 licenses will be kept), title, description, other meta data, tags, etc.
- [ ] Attach example responses to API queries that have the relevant info.

### Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by zackkrida**
Fixes #704 by @zackkrida 

## Description

Adds Nappy.co to the Catalog using the new provider DAG template in #790. This is mostly to test that PR. I've always wanted to write a provider DAG and this API was literally handwritten for us, so it was a perfect test case.

### Sample results (csv)

```csv
identifier	created_on	updated_on	ingestion_type	provider	source	foreign_identifier	foreign_landing_url	url	thumbnail	width	height	filesize	license	license_version	creator	creator_url	title	meta_data	tags	watermarked	last_synced_with_source	removed_from_source	filetype	category
002e16c4-12af-4e60-ab5f-29094e1b8060	2022-10-14 00:13:07.763443+00	2022-10-14 00:13:07.763443+00	provider_api	nappy	nappy	62	https://nappy.co/photo/62/woman-standing-by-door	https://images.nappy.co/uploads/large/201592003677l8k2kzajrojyoftnmsxtcug3qntbhbbvyto9fh26i7s6ludlgm4pevyffjnuuhr1gj1pztzsjqvfbtigdn5jf0ssynibgkny8sl7.jpg					cc0	1.0	eyeforebony	https://nappy.co/eyeforebony	Woman Standing by door	"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}"	"[{""name"": ""people_"", ""provider"": ""nappy""}, {""name"": ""building"", ""provider"": ""nappy""}, {""name"": ""person"", ""provider"": ""nappy""}, {""name"": ""human face"", ""provider"": ""nappy""}, {""name"": ""clothing"", ""provider"": ""nappy""}, {""name"": ""red"", ""provider"": ""nappy""}, {""name"": ""waist"", ""provider"": ""nappy""}, {""name"": ""street fashion"", ""provider"": ""nappy""}, {""name"": ""wall"", ""provider"": ""nappy""}, {""name"": ""shoulder"", ""provider"": ""nappy""}, {""name"": ""trousers"", ""provider"": ""nappy""}, {""name"": ""maroon"", ""provider"": ""nappy""}, {""name"": ""girl"", ""provider"": ""nappy""}, {""name"": ""fashion"", ""provider"": ""nappy""}, {""name"": ""casual dress"", ""provider"": ""nappy""}, {""name"": ""outdoor"", ""provider"": ""nappy""}, {""name"": ""woman"", ""provider"": ""nappy""}, {""name"": ""brick"", ""provider"": ""nappy""}, {""name"": ""standing"", ""provider"": ""nappy""}, {""name"": ""trouser"", ""provider"": ""nappy""}, {""name"": ""painted"", ""provider"": ""nappy""}, {""name"": ""serious"", ""provider"": ""nappy""}, {""name"": ""female"", ""provider"": ""nappy""}, {""name"": ""graffiti"", ""provider"": ""nappy""}, {""name"": ""window"", ""provider"": ""nappy""}, {""name"": ""white"", ""provider"": ""nappy""}, {""name"": ""sweater"", ""provider"": ""nappy""}, {""name"": ""necklace"", ""provider"": ""nappy""}, {""name"": ""door"", ""provider"": ""nappy""}, {""name"": ""afro"", ""provider"": ""nappy""}, {""name"": ""natural"", ""provider"": ""nappy""}, {""name"": ""hair"", ""provider"": ""nappy""}, {""name"": ""looking to the right"", ""provider"": ""nappy""}, {""name"": ""hand on hip"", ""provider"": ""nappy""}, {""name"": ""curls"", ""provider"": ""nappy""}]"	false	2022-10-14 00:13:07.763443+00	false	jpg	photograph
0045f635-d049-44f6-b8c1-f92c87b79275	2022-10-14 00:13:07.763443+00	2022-10-14 00:13:07.763443+00	provider_api	nappy	nappy	376	https://nappy.co/photo/376/woman-texting	https://images.nappy.co/uploads/large/20191208-7z8a1127-scaled-21595702861aw1ngtblnmpvgjtiye84bq9rb2fi42dadyubd5nds7tiboiuigvzfnmgdjbwstnwxbyhxtfulso2nxackh83cf95syga0jxivwcg.jpg					cc0	1.0	NappyStock	https://nappy.co/NappyStock	Woman texting	"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}"	"[{""name"": ""person"", ""provider"": ""nappy""}, {""name"": ""mobile phone"", ""provider"": ""nappy""}, {""name"": ""indoor"", ""provider"": ""nappy""}, {""name"": ""human face"", ""provider"": ""nappy""}, {""name"": ""gadget"", ""provider"": ""nappy""}, {""name"": ""communication device"", ""provider"": ""nappy""}, {""name"": ""portable communications device"", ""provider"": ""nappy""}, {""name"": ""clothing"", ""provider"": ""nappy""}, {""name"": ""wall"", ""provider"": ""nappy""}, {""name"": ""phone"", ""provider"": ""nappy""}, {""name"": ""electronic device"", ""provider"": ""nappy""}, {""name"": ""telephone"", ""provider"": ""nappy""}, {""name"": ""mobile device"", ""provider"": ""nappy""}, {""name"": ""cellphone"", ""provider"": ""nappy""}, {""name"": ""computer"", ""provider"": ""nappy""}, {""name"": ""woman"", ""provider"": ""nappy""}, {""name"": ""text"", ""provider"": ""nappy""}, {""name"": ""texting"", ""provider"": ""nappy""}, {""name"": ""message"", ""provider"": ""nappy""}, {""name"": ""office"", ""provider"": ""nappy""}, {""name"": ""yellow"", ""provider"": ""nappy""}, {""name"": ""green"", ""provider"": ""nappy""}, {""name"": ""olive"", ""provider"": ""nappy""}, {""name"": ""smiling"", ""provider"": ""nappy""}, {""name"": ""smile"", ""provider"": ""nappy""}, {""name"": ""chain"", ""provider"": ""nappy""}, {""name"": ""hair"", ""provider"": ""nappy""}]"	false	2022-10-14 00:13:07.763443+00	false	jpg	photograph
0049dfb9-19f9-4a27-a657-6f194dcededd	2022-10-14 00:13:07.763443+00	2022-10-14 00:13:07.763443+00	provider_api	nappy	nappy	3524	https://nappy.co/photo/3524/girl-with-birthday-cake	https://images.nappy.co/uploads/large/bday-2-18441639516747pa8ckqmvanydoqzigmp0uovjstcdskegxrnxp1quek64okdg3jlc6bly7nk4w1fzgqthgxr6ccafwoqugutooxaqugjfpa99thcl.jpg					cc0	1.0	alyssasieb	https://nappy.co/alyssasieb	Girl with Birthday Cake	"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}"	"[{""name"": ""people_"", ""provider"": ""nappy""}, {""name"": ""human face"", ""provider"": ""nappy""}, {""name"": ""indoor"", ""provider"": ""nappy""}, {""name"": ""smile"", ""provider"": ""nappy""}, {""name"": ""person"", ""provider"": ""nappy""}, {""name"": ""clothing"", ""provider"": ""nappy""}, {""name"": ""wall"", ""provider"": ""nappy""}, {""name"": ""floor"", ""provider"": ""nappy""}, {""name"": ""table"", ""provider"": ""nappy""}, {""name"": ""woman"", ""provider"": ""nappy""}, {""name"": ""girl"", ""provider"": ""nappy""}, {""name"": ""birthday"", ""provider"": ""nappy""}, {""name"": ""cake"", ""provider"": ""nappy""}, {""name"": ""celebrate"", ""provider"": ""nappy""}, {""name"": ""child"", ""provider"": ""nappy""}, {""name"": ""happy"", ""provider"": ""nappy""}]"	false	2022-10-14 00:13:07.763443+00	false	JPG	photograph
0090079c-5133-44c2-8e59-b31f864bdf00	2022-10-14 00:13:07.763443+00	2022-10-14 00:13:07.763443+00	provider_api	nappy	nappy	3755	https://nappy.co/photo/3755/lightbulb	https://images.nappy.co/uploads/large/stock-24-18441643628673dkiku8udymdvic76yjkeue4sknd05xqfut0auyqidws8kfwqzdezjdes3bb3zvnyseanudokyc6lgjpa436tllz8jawlirlejmr1.jpg					cc0	1.0	alyssasieb	https://nappy.co/alyssasieb	Lightbulb	"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}"	"[{""name"": ""person"", ""provider"": ""nappy""}, {""name"": ""wall"", ""provider"": ""nappy""}, {""name"": ""clothing"", ""provider"": ""nappy""}, {""name"": ""human face"", ""provider"": ""nappy""}, {""name"": ""girl"", ""provider"": ""nappy""}, {""name"": ""neutral"", ""provider"": ""nappy""}, {""name"": ""negative space"", ""provider"": ""nappy""}, {""name"": ""simple"", ""provider"": ""nappy""}, {""name"": ""blank"", ""provider"": ""nappy""}, {""name"": ""lightbulb"", ""provider"": ""nappy""}, {""name"": ""light"", ""provider"": ""nappy""}]"	false	2022-10-14 00:13:07.763443+00	false	JPG	photograph
```

## TODO

- [x] Add tests _(completed by @rwidom)_
- [x] Add popularity _(completed by @rwidom)_

## Popularity data

```js
{
  "views": 25783,
  "downloads": 1050,
  "saves": 8,
}
```

## Testing Instructions

`just recreate` and run the `nappy_workflow` through the Airflow UI. It should load `2,059` images pretty quickly.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/common/loader/provider_details.py
Comment (zackkrida) [inline openverse_catalog/dags/common/loader/provider_details.py:15]: I sorted this list alphabetically, but can revert if desired.

**Comment by zackkrida** [comment]
I'd love preliminary reviews on this, even while it's drafted, @WordPress/openverse-catalog

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (zackkrida) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:30]: Should we set an Openverse user-agent by default in `ProviderDataIngester`? That might be nice to standardize. I'll add it here shortly as well.

**Comment by zackkrida** [comment]
Ooh, and one small issue I'm observing. I'm not seeing the `thumbnail_url` being saved to the TSVs or database 🤔 Suggestions there would be appreciated!

**Comment by stacimc** [comment]
> I'm not seeing the thumbnail_url being saved to the TSVs or database 🤔 Suggestions there would be appreciated!

That's right -- we actually hardcode `thumbnail_url` to None in the [ImageStore](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/storage/image.py#L156), in favor of using the thumbnail server. I think this recently caused us trouble with another provider -- maybe SMK? I'll remove `thumbnail_url` from the suggested fields in my documentation!

**Comment by zackkrida** [comment]
@stacimc yep, that's right regarding SMK. Maybe I'll stick `thumbnail_url` in `meta_data` in case we work out a hotfix on the API side that uses that? Otherwise I can just comment it out.

**Comment by AetherUnbound** [comment]
Oh sweet, there are thumbnails available for Rawpixel too but I wasn't sure where to add them either. I'll add that to #795!

**Comment by krysal** [comment]
Let's stick with the `thumbnail_url` column, please. I don't think we need to duplicate it in the `meta_data` JSON column, we just need to remove the code for nullifying it, and then it should normally as the rest.

**Comment by AetherUnbound** [comment]
I believe the original intention was to remove the `thumbnail_url` column entirely because it was a sparsely populated field which would be better suited for the `meta_data` JSONB blob: https://github.com/WordPress/openverse-catalog/pull/526/files (similar to how we put `description` in that field as well). IIRC we were unable to actually drop that column though because the operation itself was going to take too long. This discussion appears to be across a few different places at this point though 😅 Do y'all think it makes sense to create a Make post about next steps for the `thumbnail_url` field?

**Comment by krysal** [comment]
@AetherUnbound Sorry for not responding before, my week has been somewhat sparse due to health issues, but yes, seems like a discussion in the Make blog will be necessary to decide how to save foreign thumbnails for images. I can create it if you have your hands full, just let me know!

**Comment by AetherUnbound** [comment]
=

> @AetherUnbound Sorry for not responding before, my week has been somewhat sparse due to health issues, but yes, seems like a discussion in the Make blog will be necessary to decide how to save foreign thumbnails for images. I can create it if you have your hands full, just let me know!

Thanks Krystle! It's on my TODOs, I should get to it today or the beginning of next week 🙂

**Comment by zackkrida** [comment]
@WordPress/openverse-catalog I've struggled to revisit this PR and I really don't want it to go unresolved any longer than it already has. I've unassigned myself and will add it to the agenda for next week's community meeting. I think it's really close to done and would be a great addition to the catalog.

**Comment by rwidom** [comment]
I can take a look, since I got to a pause point on https://github.com/WordPress/openverse-catalog/pull/745

**Comment by rwidom** [comment]
I'm hitting some new/weird permissions issues trying to work on this. I started by merging in main and made a new branch for that https://github.com/WordPress/openverse-catalog/tree/nappy-provider-dag-rw, but now it looks like I'm not able to push changes there either. Should I be working from a fork instead?

```
git push --set-upstream origin nappy-provider-dag-rw
Enumerating objects: 35, done.
Counting objects: 100% (35/35), done.
Delta compression using up to 8 threads
Compressing objects: 100% (24/24), done.
Writing objects: 100% (24/24), 6.35 KiB | 6.35 MiB/s, done.
Total 24 (delta 16), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (16/16), completed with 10 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/nappy-provider-dag-rw.
remote: error: Waiting on code owner review from WordPress/openverse-catalog.
To https://github.com/WordPress/openverse-catalog
 ! [remote rejected] nappy-provider-dag-rw -> nappy-provider-dag-rw (protected branch hook declined)
error: failed to push some refs to 'https://github.com/WordPress/openverse-catalog'
```

**Comment by AetherUnbound** [comment]
@rwidom How strange! It looks like we had branch protection rules for branches that match `*` (in addition to the one we have which matches `main`) which catches all branches that are at the top level (e.g. `nappy-provider-dag` but _not_ `feature/nappy-provider-dag`). Those protection rules aren't defined in our infrastructure AFAICT, so I'm going to disable them since we _also_ have the one defined for `main`. Do you mind trying to push again, first to the `nappy-provider-dag` branch and then to your copy if that fails?

**Comment by rwidom** [comment]
Thanks so much @AetherUnbound ! I wasn't able to push to this branch, though I got a different message which might have been more about my git deficits than the set up, because I _was_ able to push to https://github.com/WordPress/openverse-catalog/tree/nappy-provider-dag-rw (phew!). Dinner guest just arrived, so I can't figure out next steps right now, but at least you can see my code. Hope that's better than nothing!

**Comment by AetherUnbound** [comment]
Weird! 😮 It must be a contributor vs maintainer permissions difference 🤔 Either way, I've added your new commits to this PR! Do you think this is ready to undraft/review?

**Comment by rwidom** [comment]
Shoot, [one more commit](https://github.com/WordPress/openverse-catalog/commit/b0685b33247fc99d9a753b8ca02d523bda053d64) and it's ready.

**Comment by AetherUnbound** [comment]
Added! Oh, and looking over the PR description it seems like there are a few metrics we could use for popularity. Would you be willing to add those to the `meta_data` JSON as well? I suppose we could potentially add all 3 then decide on _which_ metric to use after the fact (so we have all available if we wanted to switch). My instinct would be to use `downloads` as the primary metric if we had to pick one though.

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:9]:
```suggestion
Notes:                  This api was written specially for Openverse.
                        There are no known limits or restrictions.
                        https://nappy.co/
```

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:30]:
```suggestion
```

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:42]:
```suggestion
```

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:55]:
```suggestion
```

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:65]: If all results are CC0, we should set this as a value on the class or instance and use it there rather than calling this function for every record!

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:67]: Since `license_info` will never be `None`, we can remove this:
```suggestion
```

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:76]: This should be a static function on the class so we can also add tests for it!

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:79]:
```suggestion
```

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:116]:
```suggestion
```

File: openverse_catalog/dags/templates/create_provider_ingester.py
Comment (AetherUnbound) [inline openverse_catalog/dags/templates/create_provider_ingester.py:1]: I think these (and the other template files) might be the result of merge conflicts, they'll need to be dropped before this is merged. (Incidentally this is also causing the weird testing failures we're seeing!)

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:38]: After running this locally, it seems like we're getting 10 images back per response. I played around with the API a bit and found that we can use the `per_page` parameter to request a different number of images! We should definitely use `batch_limit` here to speed up ingestion :smile:

```suggestion
        if not prev_query_params:
            return {
                "page": 1,
                "per_page": self.batch_limit,
            }
        else:
            return {
                **prev_query_params,
                "page": prev_query_params["page"] + 1,
            }
```

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nappy.py:91]: Thanks for adding these! We'll also want to add `downloads` [to the DDL](https://github.com/WordPress/openverse-catalog/blob/4d01826907dcf3a945e8d793783305a25cbd1ff9/docker/local_postgres/0004_openledger_image_view.sql#L11-L18) and the [image popularity metrics test](https://github.com/WordPress/openverse-catalog/blob/0728cdbe8b8ef4dc1037d877cfbf322f7252db1b/openverse_catalog/dags/common/popularity/sql.py#L44-L49)!

**Review by AetherUnbound — COMMENTED** [review]
Thanks for starting this @zackkrida and for picking this up @rwidom! I have a number of suggested changes, mostly removing some files/comment lines. I also have a suggestion about batch size & popularity-related updates.

I ran this locally and it worked great!

File: openverse_catalog/dags/templates/create_provider_ingester.py
Comment (rwidom) [reply openverse_catalog/dags/templates/create_provider_ingester.py:1]: I'm missing something here. What would need to be dropped?

File: openverse_catalog/dags/providers/provider_api_scripts/nappy.py
Comment (rwidom) [reply openverse_catalog/dags/providers/provider_api_scripts/nappy.py]: Oops! Totally, yes.

File: openverse_catalog/dags/templates/create_provider_ingester.py
Comment (rwidom) [reply openverse_catalog/dags/templates/create_provider_ingester.py:1]: OK, I tried to fix the test failures, and yeah, definitely did not succeed. I'm also getting errors about rawpixel api keys. Would love some advice!

Comment (AetherUnbound) [reply openverse_catalog/dags/templates/create_provider_ingester.py:1]: Oh, I mean to say these files should be deleted entirely 🙂 I'll go ahead and do that

**Review by AetherUnbound — COMMENTED** [review]
Whew! This is almost good to go, @rwidom would you be willing to move the `_convert_filesize` function to a static method and add tests for it as well? I can do that if that's easier 🙂

File: openverse_catalog/dags/templates/create_provider_ingester.py
Comment (rwidom) [reply openverse_catalog/dags/templates/create_provider_ingester.py]: I don't know why this comment is in the wrong place. Please forgive me!

File: tests/dags/providers/provider_api_scripts/test_nappy.py
Comment (AetherUnbound) [inline tests/dags/providers/provider_api_scripts/test_nappy.py:149]: chef's kiss

**Review by AetherUnbound — APPROVED** [review]
Excellent, thank you @rwidom and @zackkrida!

**Review by rwidom — APPROVED** [review]
I think these are ready, but do we need someone else to sign off, @zackkrida ?

**Comment by rwidom** [comment]
> I think these are ready, but do we need someone else to sign off, @zackkrida ?

Oops, I guess not and that's why you removed the request from Staci! :)

**Comment by zackkrida** [comment]
@rwidom I removed the Staci request because I was going to review the PR. I have, and I think this is ready to merge! Thank you so much again for finishing the PR.

**Pull request merged**
